apps/dsb-message-broker/src/channel/dto/create-channel.dto.ts

```diff
@@ -1,52 +1,75 @@
 import { IsNotEmpty, IsString, IsOptional } from 'class-validator';
 import { ApiProperty } from '@nestjs/swagger';
+import { JSONSchemaType } from 'ajv';
 
 export class CreateChannelDto {
     @ApiProperty({
-        type: String,
-        description: 'Fully qualified channel name (fcqn)',
-        example: 'test.channels.testapp.apps.testorganization.iam.ewc'
+        type: 'string',
+        required: true,
+        format: '{channel_name}.channels.{app_name}.apps.{organization_name}.iam.ewc',
+        description: 'Fully Qualified Channel Name (fcqn)'
     })
     @IsString()
     @IsNotEmpty()
     fqcn: string;
 
     @ApiProperty({
-        type: Object,
-        description: 'Array of topics that is available in the channel'
+        type: 'object',
+        properties: {
+            namespace: {
+                type: 'string'
+            },
+            schema: {
+                type: 'string | JSONSchemaType'
+            }
+        },
+        required: false,
+        description: 'Array of topic objects that determines topics for messages.'
     })
     @IsOptional()
     topics?: {
         namespace: string;
-        schemaType: 'json' | 'xml' | 'JSON' | 'XML';
-        schema: string;
+        schema: JSONSchemaType<any> | string;
     }[];
 
     @ApiProperty({
-        type: Array,
+        isArray: true,
+        required: false,
         description:
-            'Array of DIDs and roles that have permision to publish messages to the channel. If it is ommited, any user with "user" role can publish messages to the channel.'
+            'Array of DIDs that have permision to edit the channel. If it is ommited, creator of the channel will be the default admin.'
+    })
+    @IsOptional()
+    admins?: string[];
+
+    @ApiProperty({
+        isArray: true,
+        required: false,
+        description:
+            'A mixed array of DIDs and roles that have permision to publish messages to the channel. If it is ommited, any user with "user" role can publish messages to the channel.'
     })
     @IsOptional()
     publishers?: string[];
 
     @ApiProperty({
-        type: Array,
+        isArray: true,
+        required: false,
         description:
-            'Array of DIDs and roles that have permision to subscribe to the channel. If it is ommited, any user with "user" role can subscribe to the channel.'
+            'A mixed array of DIDs and roles that have permision to subscribe to the channel. If it is ommited, any user with "user" role can subscribe to the channel.'
```

Comment:
we probably want an example (also for the other properties too)

Reply:
added